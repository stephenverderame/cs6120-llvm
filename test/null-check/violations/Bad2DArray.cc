// NOLINTBEGIN
#include <cassert>
#include <cstring>
int main(int argc, char** argv)
{
    if (argc >= 128) {
        return 1;
    }
    char words[128][128];

    for (int i = 0; i < argc; ++i) {
        const auto word = (argv + i);
        assert(word != nullptr);
        const auto word_data = *word;
        const auto word_len = strlen(word_data);
        for (int j = 0; j < word_len; ++j) {
            const auto c = word_data + j;
            assert(c != nullptr);
            words[i][j] = *c;
        }
    }
}

// NOLINTEND