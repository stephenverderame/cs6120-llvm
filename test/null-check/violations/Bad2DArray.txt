[31mSafety Violation[00m: Use of potentially null pointer in store at Bad2DArray.cc:19
Because an index could not be proven to be in range, the following counterexample was found:
	[1margc[0m = 1
	[1mi[0m = 1
	[1mj[0m = 16385
