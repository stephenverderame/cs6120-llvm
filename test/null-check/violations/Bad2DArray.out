[31mSafety Violation[00m: Use of potentially null pointer in store at Bad2DArray.cc:19
